first-react-app/src/components/SubmitForm.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import CheckboxInput from "./CheckboxInput";
+import DateInput from "./DateInput";
+import RadioInput from "./RadioInput";
+import SelectInput from "./SelectInput";
+import TextInput from "./TextInput";
+import UploadInput from "./UploadInput";
+
+export interface IFormData {
+  (a: string[]): void;
+}
+
+interface ISubmit extends Object {
+  isFormCorrect: IFormData;
+}
+
+class SubmitForm extends React.Component<ISubmit, { popUpClass: string; popUpText: string }> {
+  textRef: React.RefObject<TextInput>;
+  dateRef: React.RefObject<DateInput>;
+  selectRef: React.RefObject<SelectInput>;
+  checkboxRef: React.RefObject<CheckboxInput>;
+  radioRef: React.RefObject<RadioInput>;
+  uploadRef: React.RefObject<UploadInput>;
+  constructor(props: ISubmit) {
+    super(props);
+    this.state = {
+      popUpClass: "error-hidden",
+      popUpText: "You have errors in form. Please correct them.",
+    };
+    this.textRef = React.createRef();
+    this.dateRef = React.createRef();
+    this.selectRef = React.createRef();
+    this.checkboxRef = React.createRef();
+    this.radioRef = React.createRef();
+    this.uploadRef = React.createRef();
+    this.handleErrors = this.handleErrors.bind(this);
+  }
+
+  handleErrors() {
+    const isTextWrong = this.textRef.current?.state.hasError;
+    const isDateWrong = this.dateRef.current?.state.hasError;
+
+    if (isTextWrong || isDateWrong) {
+      this.setState({
+        popUpClass: "error-visible",
+        popUpText: "You have errors in form. Please correct them.",
+      });
+    } else {
+      this.setState({
+        popUpClass: "confirm-visible",
+        popUpText: "You have successfully created a card",
+      });
+
+      const checkBoxValuetoString = this.checkboxRef.current!.state.checked as boolean;
+
+      const formData = [
+        this.textRef.current?.state.text as string,
+        this.dateRef.current?.state.date as string,
+        this.selectRef.current?.state.rating as string,
+        checkBoxValuetoString.toString() as string,
+        this.radioRef.current?.state.whoMade as string,
+        this.uploadRef.current?.state.pic as string,
+      ];
```

Comment:
Overall, this is bad, you are reading state via ref from the components you created.
Child components should not have direct access to the state of their parent components, and parent components should not have direct access to the state of their child components.